### PR TITLE
Update DG's use cases and user stories

### LINKED ISSUE #226: DG: Amend use cases and user stories

### PR DESCRIPTION
**Pull request by gabztcr**
Fixes #226 + amend UG to include the ability to empty an optional field

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -413,380 +413,637 @@ The following proposed sequence diagram shows how the updating of the OCP would
...
+| `* *`      | Product Manager                            | add contacts and location under my meetings                                 | know who is involved and where the meeting is taking place |
+| `* *`      | Product Manager with many meetings         | search for specific ones                                                    | easily find them from my entire list                       |
+| `* *`      | Product Manager with many meetings         | have my meetings sorted chronologically by its start time                   | look out for the earlier meetings that were scheduled      |
+| `* *`      | Product Manager                            | be informed when a meeting is close to starting, on-going or has ended      | be updated on the status of my meetings                    |
+| **EPIC C** | Product Manager                            | organise my developer or stakeholder contacts                               | reference them easily                                      |
+| `* * *`    | Product Manager                            | add contacts related to the product (developers/stakeholders)               | store their details for future communication               |
```
Comment (chrystalquek) [inline docs/DeveloperGuide.md:443]: i think the (developers/stakeholders) not necessary, stated in other user story

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -413,380 +413,637 @@ The following proposed sequence diagram shows how the updating of the OCP would
...
+| `* *`      | Product Manager                            | see a calendar view of my deliverables and meetings                         | know which product-related events are upcoming and urgent  |
+| `*`        | Product Manager                            | toggle between daily, weekly and monthly view of the calendar               | have a variety of views to see my upcoming events          |
+| **EPIC E** | forgetful or inexperienced Product Manager | view app instructions and tips                                              | be able to use it as intended                              | 
+| `* * *`    | Product Manager                            | receive feedback from the app                                               | know the system has successfully registered my action      |
+| `* *`      | Product Manager                            | view a helpful popup that directs me to the app's user guide                | easily access the relevant instructions                    |
```
Comment (chrystalquek) [inline docs/DeveloperGuide.md:455]: 'popup' might contain an implementation detail

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -413,380 +413,637 @@ The following proposed sequence diagram shows how the updating of the OCP would
...
+     * 2a1. Productiv displays an error message.
+     * 2a2. User enters the command again.
+     * 2a3. Steps 2a1-2a2 are repeated until the command entered is correct.
+     
+       Use case resumes from step 3.
+       
+### Deliverable
 
-**Use case: `UC10 - Add a contact`**
+**Use case: UC11 - Add a deliverable**
```
Comment (chrystalquek) [inline docs/DeveloperGuide.md:521]: numbering

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -413,380 +413,637 @@ The following proposed sequence diagram shows how the updating of the OCP would
...
-* **`User is in the Contact mode`**
+* User is in the deliverable mode.
+
+**Guarantee(s):**
+* Upon successful command, the OCP will be updated accordingly.
 
 **MSS**
+1. User chooses to add a deliverable.
+2. User enters the command to add a deliverable into the command box.
+3. Productiv adds the deliverable into the deliverable list.
```
Comment (chrystalquek) [inline docs/DeveloperGuide.md:533]: View Panel
(can choose not to add)

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -413,380 +413,637 @@ The following proposed sequence diagram shows how the updating of the OCP would
...
+**Use case: UC15 - View a deliverable**
 
+**Precondition(s):**
+* User is in the deliverable mode.
+* Deliverable to view exists.
 
+**MSS**
+1. User chooses to view a deliverable.
+2. User enters the command to view a deliverable into the command box.
+3. Productiv displays the deliverable in the view box.
```
Comment (chrystalquek) [inline docs/DeveloperGuide.md:622]: right panel? view panel? view box? im not too sure either, but i used right panel

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -413,380 +413,637 @@ The following proposed sequence diagram shows how the updating of the OCP would
...
 
 **Precondition(s):**
-* **`User is in the Deliverable mode`** 
+* User is in the deliverable mode.
+* User has keyword(s) for Productiv to find deliverables with.
 
 **MSS**
+1. User chooses to find deliverables.
+2. User enters the command and keyword(s) to find deliverables into the command box.
+3. Productiv returns all matching deliverables in the list box.
```
Comment (chrystalquek) [inline docs/DeveloperGuide.md:643]: same, i think its left panel

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -413,380 +413,637 @@ The following proposed sequence diagram shows how the updating of the OCP would
...
+      
+**Use case: UC18 - Delete a deliverable**
 
-* 2a. Produtiv detects an error in the command.
+**Precondition(s):**
+* User is in the deliverable mode.
+* Deliverable to delete exists.
+
+**Guarantee(s):** 
+* Upon successful command, the OCP will be updated accordingly.
```
Comment (chrystalquek) [inline docs/DeveloperGuide.md:683]: actually if you're gonna say this, probably should say calendar will also be updated

File: docs/UserGuide.md [hunk 8]
```diff
@@ -55,17 +55,17 @@ Here is a sequence of example commands you can try:<br>
...
-
 * Field-parameter pairs in square brackets are optional.<br>
   e.g `n/NAME [p/PHONE]` can be used as `n/Jason p/98890112` or as `n/Jason`.
 
 * Field-parameter pairs can be in any order.<br>
   e.g. if the command specifies `n/NAME e/EMAIL`, `e/EMAIL n/NAME` is also acceptable.
   
 * If multiple and/or repeat parameters are provided for the same field, only the last parameter will be accepted.<br>
-  e.g. if you input the command `r/dev n/NAME r/stk e/EMAIL r/stk` it will be accepted as `n/NAME r/stk e/EMAIL`.
+  e.g. if you input the command `r/dev n/NAME r/stk e/EMAIL r/stk`, it will be accepted as `n/NAME r/stk e/EMAIL`.
```
[the remark this answers is not in the document]
Comment (gabztcr) [reply docs/UserGuide.md:65]: added `add` into the commands

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (gabztcr) [reply docs/DeveloperGuide.md:521]: use G01, M01, D01, C01 ...

File: docs/UserGuide.md [hunk 9]
```diff
@@ -197,7 +196,7 @@ Examples:
 *  `edit 1 d/Must include username, email and password fields by/15-12-2020 12:00`
 edits the description of the 1st deliverable to be `Must include username, email and password fields`
 and its deadline to be `15-12-2020 12:00`.
-*  `edit 2 c/Betsy Crowe, Jeremey` edits the contacts of the 2nd deliverable to be `Betsy Crowe, Jeremey`.
```
Comment (chrystalquek) [inline docs/UserGuide.md:200]: no harm keeping

**Review by chrystalquek — APPROVED** [review]
LGTM, consider moving some of UI changes found under Guarantees to subsequent steps in the MSS

File: docs/UserGuide.md (hunk 9 above)
Comment (gabztcr) [reply docs/UserGuide.md:200]: not very in favour of cluttering the UG/MSS steps. can discuss again with the rest during meeting

**Pull request merged**